src/ui/CategoryFacet/CategoryFacet.ts

```diff
@@ -235,22 +346,82 @@ export class CategoryFacet extends Component {
     return this.categoryValueRoot.children;
   }
 
+  private renderValues(categoryFacetResult: ICategoryFacetResult, numberOfRequestedValues: number) {
```

Comment:
This method is in general pretty hard to read. 

I'd break it down into some smaller private methods, for readability.